WallsAndHoles/tilemap.cpp

```diff
@@ -6,21 +6,26 @@ TileMap::TileMap(QSize mapSize,
                  QObject *parent)
     : QObject(parent)
     , mMap(mapSize.width(), mapSize.height())
+    , mDefaultTileTemplateSet(new TileTemplateSet("Map Tile Templates", this))
 {
     for (int x = 0; x < mMap.size().width(); ++x) {
         for (int y = 0; y < mMap.size().height(); ++y) {
             mMap(x, y) = QSharedPointer<Tile>::create(nullptr, x, y, this);
             connect(mMap(x, y).data(), &Tile::tileChanged,
                     this, &TileMap::tileChanged);
+            connect(mMap(x, y).data(), &Tile::tilePinged,
+                    this, &TileMap::tilePinged);
         }
     }
 
-
     // tileChanged() and resized() signals should always be followed by a mapChanged() signal
-    connect(this, &TileMap::tileChanged, this, [this] () { emit mapChanged(); });
-    connect(this, &TileMap::resized, this, [this] () { emit mapChanged(); });
-}
+    connect(this, &TileMap::tileChanged, this, &TileMap::mapChanged);
+    connect(this, &TileMap::resized, this, &TileMap::mapChanged);
 
+    //set up default tile templates. TODO this should be impacted by inital map properties.
+    mDefaultTileTemplateSet->addTileTemplate(nullptr); //For an eraser
```

Comment:
Impaceted ... Properties was just that one may changed the height of the default wall or something of that sort. Ill look into having an empty template, though I think it may produce more work than it saves...